Sources/JWTKit/JWK/JWK.swift

```diff
@@ -79,26 +107,63 @@ public struct JWK: Decodable {
     ///
     /// The `kid` value is a case-sensitive string.
     public var keyIdentifier: JWKIdentifier?
-    
+
+    // RSA keys
+    // Represented as the base64url encoding of the value’s unsigned big endian representation as an octet sequence.
     /// `n` Modulus.
     public var modulus: String?
-    
+
     /// `e` Exponent.
     public var exponent: String?
-    
+
     /// `d` Private exponent.
     public var privateExponent: String?
-    
+
+    // ECDSA keys
+    public var x: String?
+
+    public var y: String?
+
     private enum CodingKeys: String, CodingKey {
         case keyType = "kty"
         case algorithm = "alg"
         case keyIdentifier = "kid"
         case modulus = "n"
         case exponent = "e"
         case privateExponent = "d"
+        case x
+        case y
     }
 
     public init(json: String) throws {
         self = try JSONDecoder().decode(JWK.self, from: Data(json.utf8))
     }
+    
+    public init(rsaKeyUsing algorithm: Algorithm?, keyIdentifier: JWKIdentifier?, n: String?, e: String?, d: String?) {
```

Comment:
What about static methods here? Let's use the full names of properties here, too.

```swift
static func rsa(_ algorithm:identifier:modulus:...)
```